workflow-service/src/main/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowExecutionFactory.java

```diff
@@ -0,0 +1,66 @@
+package com.redhat.parodos.workflow.execution.aspect;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import com.redhat.parodos.workflow.context.WorkContextDelegate;
+import com.redhat.parodos.workflow.definition.entity.WorkFlowDefinition;
+import com.redhat.parodos.workflow.execution.continuation.WorkFlowContinuationServiceImpl;
+import com.redhat.parodos.workflow.execution.repository.WorkFlowRepository;
+import com.redhat.parodos.workflow.execution.scheduler.WorkFlowSchedulerServiceImpl;
+import com.redhat.parodos.workflow.execution.service.WorkFlowServiceImpl;
+import com.redhat.parodos.workflows.work.WorkContext;
+import org.springframework.stereotype.Service;
+
+@Service
+public class WorkFlowExecutionFactory {
+
+	private final WorkFlowServiceImpl workFlowService;
+
+	private final WorkFlowRepository workFlowRepository;
+
+	private final WorkFlowSchedulerServiceImpl workFlowSchedulerService;
+
+	private final WorkFlowContinuationServiceImpl workFlowContinuationServiceImpl;
+
+	public WorkFlowExecutionFactory(WorkFlowServiceImpl workFlowService, WorkFlowRepository workFlowRepository,
+			WorkFlowSchedulerServiceImpl workFlowSchedulerService,
+			WorkFlowContinuationServiceImpl workFlowContinuationServiceImpl) {
+		this.workFlowService = workFlowService;
+		this.workFlowRepository = workFlowRepository;
+		this.workFlowSchedulerService = workFlowSchedulerService;
+		this.workFlowContinuationServiceImpl = workFlowContinuationServiceImpl;
+	}
+
+	public WorkFlowExecutionInterceptor createExecutionHandler(WorkFlowDefinition definition, WorkContext workContext) {
```

Comment:
if the words Handler and Interceptor are interchangeable it is better to just use one of them. I was looking for a Handler Class but didn't find one.

Reply:
mmm okay there are Handlers..

Reply:
still since they are not of the same type and the return type is an interceptor  createExecutionInterceptor is natural

Reply:
I agree. I battled with naming more than once here :-) 
Consistency is a virtue. I'll stick to one.